### ISSUE TITLE
[Bug]: SSL did not renew, even after restart

### ISSUE THREAD

**Issue by merrcury** [appsmithorg/appsmith#29497]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

We have been using Appsmith for over 6 months but for some reason SSL did not renew itself this time. Its not getting renewed even after restarting docker. 

### Steps To Reproduce

It happened automatically

### Environment

Production

### Version

Self Hosted - Appsmith Community v1.9.50

**Comment by sharat87** [comment]
Hey @merrcury, are you using `docker-compose` to run Appsmith?

If yes, can you run the following command please?

```
sudo docker-compose exec appsmith /opt/appsmith/renew-certificate.sh
```

That should trigger a renewal of the certificate. Please confirm if that helps.

**Comment by merrcury** [comment]
Yes, I am using `docker-compose`
I got below error
``` bash
OCI runtime exec failed: exec failed: unable to start container process: exec: "/opt/appsmith/renew-certificate.sh": stat /opt/appsmith/renew-certificate.sh: no such file or directory: unknown
```

**Comment by sharat87** [comment]
@merrcury, sorry about that. Can you share the logs of your container please? You can get this by running `docker-compose logs appsmith > logs.txt` and then sharing the `logs.txt` file.

**Comment by merrcury** [comment]
Here you go, It might have less logs because I restarted docker in the morning
[logs.txt](https://github.com/appsmithorg/appsmith/files/13634405/logs.txt)

**Comment by sharat87** [comment]
Hey @merrcury, thanks for the logs, that helped. Your instance looks to be up now, please check and confirm.

Here's a brief of what I think happened here.

1. The TLS cert you were using is set to expire at `Dec  8 23:05:02 2023 GMT`, as seen from line 137 in the logs.
2. The auto-renewal system in Appsmith relies on `certbot`, which recently changed it's non-interactive operation. This is the reason renewal was failing for you.
3. We've noticed that problem in certbot and fixed that problem by switching to Caddy in `v1.9.51`. Along with this, we removed the `renew-certificate.sh` script, as it's no longer needed. That's why you an error you run that command.
4. However, now that you've run the latest version, that uses Caddy for certificate management, we can see from the logs that Caddy has indeed provisioned a new cert and has started using it.

Line 166 in logs:
```json
{"level":"info","ts":1702288831.4396663,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["console.rocketium.com"]}
```

**Comment by merrcury** [comment]
@sharat87 Thanks for the quick turnaround. This made sense & Our server is came back up. I am closing this issue.

**Comment by merrcury** [comment]
Hey @sharat87, Now Its down for more than 24 hours. Is there anything I should be aware of? It not even coming back up.

**Comment by sharat87** [comment]
Hey @merrcury, sorry for the trouble. Can you share the logs again please?
Please find some time at https://cal.com/shrismith/i29497 so we can get on a call and troubleshoot. Thanks!

**Comment by merrcury** [comment]
Hey @sharat87 I wasn't well. PFA docker logs. I have also booked a slot at 16:00 IST
[logs.txt](https://github.com/appsmithorg/appsmith/files/13724154/logs.txt)

**Comment by sharat87** [comment]
Hey @merrcury, can you pull the latest version again and try please?Just looked at the logs, this is a known problem fixed in `v1.9.56` itself. Current latest is `v1.9.57`, which should also be fine for you.

**Comment by sharat87** [comment]
Thanks for the call. It appears Watchtower wasn't updating Appsmith for some reason, and when we did a `docker-compose pull`, the latest version was downloaded and Appsmith was immediately up and running.

On the release of the next version of Appsmith, if you still notice that Appsmith isn't auto-updating, please reach out again on support@appsmith.com. Thank you.

I'm closing this now. Thanks!
